trio/_socket.py

```diff
@@ -510,26 +511,213 @@ async def _resolve_address_nocp(
     if family == _stdlib_socket.AF_INET6:
         list_normed = list(normed)
         assert len(normed) == 4
-        # typechecking certainly doesn't like this logic, but given just how broad
-        # Address is, it's quite cumbersome to write the below without type: ignore
         if len(address) >= 3:
-            list_normed[2] = address[2]  # type: ignore
+            list_normed[2] = address[2]
         if len(address) >= 4:
-            list_normed[3] = address[3]  # type: ignore
-        return tuple(list_normed)  # type: ignore
+            list_normed[3] = address[3]
+        return tuple(list_normed)
     return normed
 
 
-# TODO: stopping users from initializing this type should be done in a different way,
-# so SocketType can be used as a type. Note that this is *far* from trivial without
-# breaking subclasses of SocketType. Can maybe add abstract methods to SocketType,
-# or rename _SocketType.
 class SocketType:
-    def __init__(self) -> NoReturn:
-        raise TypeError(
-            "SocketType is an abstract class; use trio.socket.socket if you "
-            "want to construct a socket object"
-        )
+    def __init__(self) -> None:
+        if type(self) == SocketType:
+            raise TypeError(
+                "SocketType is an abstract class; use trio.socket.socket if you "
+                "want to construct a socket object"
+            )
+
+    def detach(self) -> int:
+        raise NotImplementedError
+
+    def fileno(self) -> int:
+        raise NotImplementedError
+
+    def getpeername(self) -> AddressFormat:
+        raise NotImplementedError
+
+    def getsockname(self) -> AddressFormat:
+        raise NotImplementedError
+
+    @overload
+    def getsockopt(self, /, level: int, optname: int) -> int:
+        ...
+
+    @overload
+    def getsockopt(self, /, level: int, optname: int, buflen: int) -> bytes:
+        ...
+
+    def getsockopt(
+        self, /, level: int, optname: int, buflen: int | None = None
+    ) -> int | bytes:
+        raise NotImplementedError
+
+    @overload
+    def setsockopt(self, /, level: int, optname: int, value: int | Buffer) -> None:
+        ...
+
+    @overload
+    def setsockopt(self, /, level: int, optname: int, value: None, optlen: int) -> None:
+        ...
+
+    def setsockopt(
+        self,
+        /,
+        level: int,
+        optname: int,
+        value: int | Buffer | None,
+        optlen: int | None = None,
+    ) -> None:
+        raise NotImplementedError
+
+    def listen(self, /, backlog: int = min(_stdlib_socket.SOMAXCONN, 128)) -> None:
+        raise NotImplementedError
+
+    def get_inheritable(self) -> bool:
+        raise NotImplementedError
+
+    def set_inheritable(self, inheritable: bool) -> None:
+        raise NotImplementedError
+
+    if sys.platform == "win32" or (
+        not TYPE_CHECKING and hasattr(_stdlib_socket.socket, "share")
+    ):
+
+        def share(self, /, process_id: int) -> bytes:
```

Comment:
This coverage warning here also doesn't make sense given we're running the tests on Windows...?

Reply:
Yeah no clue :<

Reply:
on a rerun there were no coverage warning, so seems like a temporary glitch.